metldata/load/load.py

```diff
@@ -48,10 +51,73 @@ async def load_artifacts_using_dao(
 
+    (
+        removed_resource_tags,
+        new_resources,
+        changed_resources,
+    ) = await _get_changed_resources(
+        artifact_resources=artifact_resources,
+        artifact_info_dict=artifact_info_dict,
+        dao_collection=dao_collection,
+    )
+
+    await process_removed_resources(
+        resource_tags=removed_resource_tags, dao_collection=dao_collection
+    )
+
+    await process_new_or_changed_resources(
+        resources=new_resources, dao_collection=dao_collection
+    )
+
+    await process_new_or_changed_resources(
+        resources=changed_resources, dao_collection=dao_collection
+    )
+
+
+async def _get_changed_resources(  # pylint: disable=too-many-locals
+    artifact_resources: ArtifactResourceDict,
+    artifact_info_dict: dict[str, ArtifactInfo],
+    dao_collection: ArtifactDaoCollection,
+) -> tuple[set[str], dict[str, ArtifactResource], dict[str, ArtifactResource]]:
+    """Extract changed resources using DAOs and currently submitted artifacts
+
+    Returns the following collections in order:
+     - A set of resource tags for artifact resources that have been removed
+     - A dict of new artifact resources indexed by the corresponding resource_tag
+     - A dict of changed artifact resources indexed by the corresponding resource_tag
+    """
+
+    # only need to collect resource tag to check if something needs to be deleted
+    seen_resource_tags = []
+
+    # collect new/changed resources indexed by their resource_tag for insert/upsert
+    new_resources = {}
+    changed_resources = {}
+
+    existing_resource_tags = await dao_collection.get_all_resource_tags()
+
     for artifact_name, artifact_contents in artifact_resources.items():
         for artifact_content in artifact_contents:
-            await load_artifact_resources(
+            resources = extract_all_resources_from_artifact(
                 artifact_content=artifact_content,
                 artifact_info=artifact_info_dict[artifact_name],
-                dao_collection=dao_collection,
             )
+
+            # check for each resource if it does already exist and is changed
+            for resource in resources:
+                resource_tag = f"{artifact_name}#{resource.class_name}#{resource.id_}"
+                seen_resource_tags.append(resource_tag)
+
+                if resource_tag not in existing_resource_tags:
+                    new_resources[resource_tag] = resource
+                else:
+                    dao = await dao_collection.get_dao(
+                        artifact_name=artifact_name, class_name=resource.class_name
+                    )
+                    old_resource = await dao.get_by_id(resource.id_)
+                    if old_resource != resource:
+                        changed_resources[resource_tag] = resource
+
+    removed_resource_tags = existing_resource_tags - set(seen_resource_tags)
```

Comment:
```suggestion
    existing_resource_tags.difference(seen_resource_tags)
```

Reply:
Instead of tracking all `seen_resource_tags`, you could also track the `unchanged_resource_tags`, and then set
```
removed_resource_tags = existing_resource_tags.difference(changed_resources, unchanged_resource_tags, new_resources)
```